cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1922,19 +1916,15 @@ func (p *BicepProvider) ensureParameters(
 				configuredParameters[key] = azure.ArmParameterValue{
 					Value: value,
 				}
-				configuredParameters[prompt.key] = azure.ArmParameterValue{
-					Value: value,
-				}
 			}
 		}
 	}
 
 	if configModified {
 		if err := p.envManager.Save(ctx, p.env); err != nil {
-			p.console.Message(ctx, fmt.Sprintf("warning: failed to save configured values: %v", err))
```

Comment:
In case it isn't clear why ignoring this error may not be ideal:

Suppose that `Save()` failed due to flakiness once -- this variable ends up getting re-prompted which is confusing behavior to a user. Straight up failing is better here.

A code [demonstration](https://github.com/Azure/azure-dev/compare/main...weikanglim:weilim/rm-param-cache-failsave-demo?expand=1).

Here's it live:
![Screenshot 2024-09-20 at 1 22 38 PM](https://github.com/user-attachments/assets/82f3967d-a088-4618-ad0e-309b3a64d488)